hrabsence/CRM/HRAbsence/Test/Fabricator/HRAbsenceType.php

```diff
@@ -15,9 +29,33 @@ public static function fabricate($params = []) {
       $params['name'] = 'Absence Type ' . microtime();
     }
 
-    $params = array_merge(self::$defaultParams, $params);
-
-    return CRM_HRAbsence_BAO_HRAbsenceType::create($params);
+    return array_merge(self::$defaultParams, $params);
+  }
+  
+  /**
+   * Absence Type fabricator method, uses BAO directly
+   * 
+   * @param array $params
+   *   Parameters to be passed to BAO to create absence type
+   * @return CRM_HRAbsence_BAO_HRAbsenceType
+   */
+  public static function fabricate($params = []) {
+    return CRM_HRAbsence_BAO_HRAbsenceType::create(self::buildParamsArray($params));
+  }
+  
+  /**
+   * Absence Type fabricator method that uses API and returns values of created
+   * record.
+   * 
+   * @param array $params
+   *   Parameters to be passed to API call to create absence type
+   * @return array
+   *   Values of created absence type
+   */
+  public static function fabricateUsingAPI($params = []) {
```

Comment:
If it's just a matter of the values returned, why not update the fabricate() method to return the values you need? Seems to be simpler than adding a new method (also, API calls are slow and will impact tests performance). You can change the fabricate method to something like this:

```php
public static function fabricate($params = []) {
   $absenceType = CRM_HRAbsence_BAO_HRAbsenceType::create(self::buildParamsArray($params));
   $absenceType = CRM_HRAbsence_BAO_HRAbsenceType::findByPk($absenceType->id);
   return $absenceType;
}
```

or, you can even make the findByPk part optional (to avoid unnecessary sql queries):

```php
public static function fabricate($params = [], $loadAfterSave = false) {
   $absenceType = CRM_HRAbsence_BAO_HRAbsenceType::create(self::buildParamsArray($params));
   
   if($loadAfterSave) {
     $absenceType = CRM_HRAbsence_BAO_HRAbsenceType::findByPk($absenceType->id);
   }
   
   return $absenceType;
}
```